conversion/resource_generation.go

```diff
@@ -0,0 +1,413 @@
+			Connectivity: &datastreampb.ConnectionProfile_StaticServiceIpConnectivity{},
+		},
+		ValidateOnly: createResourceData.ConnectionProfile.ValidateOnly,
+	}
+
+	// If destination source profile is to be created, create a gcs bucket first
+	var bucketName string
+	if !createResourceData.ConnectionProfile.IsSource {
+		bucketName = strings.ToLower("GCS-" + createResourceData.ConnectionProfile.Id)
+		err = utils.CreateGCSBucket(bucketName, createResourceData.ConnectionProfile.ProjectId, createResourceData.ConnectionProfile.Region)
+		if err != nil {
+			createResourceData.Error = err
+			return common.TaskResult[*ConnectionProfileReq]{Result: createResourceData, Err: err}
+		}
+	}
+	createResourceData.ConnectionProfile.BucketName = bucketName
+
+	// Set Profile for resource creation
+	r.setConnectionProfileFromRequest(createResourceData, req)
+
+	// Create or Validate Resource
+	op, err := dsClient.CreateConnectionProfile(createResourceData.ctx, req)
+	if err != nil {
+		createResourceData.Error = err
+		return common.TaskResult[*ConnectionProfileReq]{Result: createResourceData, Err: err}
+	}
+	_, err = op.Wait(createResourceData.ctx)
+	if err != nil {
+		createResourceData.Error = err
+		return common.TaskResult[*ConnectionProfileReq]{Result: createResourceData, Err: err}
+	}
+
+	if !createResourceData.ConnectionProfile.ValidateOnly {
+		fmt.Printf("Connection Profile for Datashard id %v has been created: %v\n", createResourceData.ConnectionProfile.DatashardId, createResourceData.ConnectionProfile.Id)
+		// In case of failure, add resources to be cleaned up
+		resourcesForCleanup = append(resourcesForCleanup, createResourceData)
+	} else {
+		fmt.Printf("Connection Profile for Datashard id %v has been validated: %v\n", createResourceData.ConnectionProfile.DatashardId, createResourceData.ConnectionProfile.Id)
+	}
+
+	return common.TaskResult[*ConnectionProfileReq]{Result: createResourceData, Err: nil}
+}
+
+// Sets Profile for resource creation
+func (r ResourceGenerationStruct) setConnectionProfileFromRequest(details *ConnectionProfileReq, req *datastreampb.CreateConnectionProfileRequest) error {
+	if details.ConnectionProfile.IsSource {
+		port, err := strconv.ParseInt((details.ConnectionProfile.Port), 10, 32)
+		if err != nil {
+			return err
+		}
+		req.ConnectionProfile.Profile = &datastreampb.ConnectionProfile_MysqlProfile{
+			MysqlProfile: &datastreampb.MysqlProfile{
+				Hostname: details.ConnectionProfile.Host,
+				Port:     int32(port),
+				Username: details.ConnectionProfile.User,
+				Password: details.ConnectionProfile.Password,
+			},
+		}
+	} else {
+		req.ConnectionProfile.Profile = &datastreampb.ConnectionProfile_GcsProfile{
+			GcsProfile: &datastreampb.GcsProfile{
+				Bucket:   details.ConnectionProfile.BucketName,
+				RootPath: "/",
+			},
+		}
+		return nil
+	}
+	return nil
+}
+
+func (r ResourceGenerationStruct) GetSpannerRegion(ctx context.Context, projectId string, instanceName string) (string, error) {
```

Comment:
Nit: Add docs for all public methods. 

Nit: Move public methods to the top of the file and the private methods to the bottom.